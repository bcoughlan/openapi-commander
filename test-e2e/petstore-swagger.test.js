const fs = require('fs/promises')
const express = require('express')
const { GeneratorTest } = require('./test-utils')

describe('Petstore Swagger 2', () => {
  
  let generatorTest
  beforeAll(async () => {
    generatorTest = await GeneratorTest('test-resources/petstore-swagger.yaml')
  })

  afterAll(async () => {
    await generatorTest.cleanup()
  })

  it('lists categories', async () => {
    expect(await generatorTest.run('')).toMatchSnapshot()
  })

  it('lists commands', async () => {
    expect(await generatorTest.run('pets')).toMatchSnapshot()
  })

  it('debug prints', async () => {
    expect(await generatorTest.run('-d pets listPets')).toMatchSnapshot()
  })

})
