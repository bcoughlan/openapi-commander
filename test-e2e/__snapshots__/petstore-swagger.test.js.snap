// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Petstore Swagger 2 debug prints 1`] = `
{
  "stderr": "",
  "stdout": "GET http://petstore.swagger.io/v1/pets
Accept: */*
",
}
`;

exports[`Petstore Swagger 2 lists categories 1`] = `
{
  "code": 1,
  "stderr": "",
  "stdout": "Usage: petstore-swagger [options] [command]

Options:
  -p, --print <mode>     Print the HTTP request instead of sending it.
                         (choices: "curl", "plain")
  -v, --verbose          Includes the response headers in the output
  -s, --server <server>  Base URL to use for requests
  -a, --auth <auth>      Authorization header to send
  -c, --accept <accept>  Accept header to send
  -h, --help             display help for command

Commands:
  pets
  help [command]         display help for command
",
}
`;

exports[`Petstore Swagger 2 lists commands 1`] = `
{
  "code": 1,
  "stderr": "",
  "stdout": "Usage: petstore-swagger pets [options] [command]

Options:
  -h, --help           display help for command

Commands:
  listPets [options]   List all pets
  createPets           Create a pet
  showPetById <petId>  Info for a specific pet
  help [command]       display help for command
",
}
`;
